tests/test_sync.py

```diff
@@ -29,18 +29,44 @@ def sync_function():
     end = time.monotonic()
     assert result == 42
     assert end - start >= 1
+
     # Set workers to 1, call it twice and make sure that works right
-    loop = asyncio.get_event_loop()
-    old_executor = loop._default_executor
-    loop.set_default_executor(ThreadPoolExecutor(max_workers=1))
-    try:
-        start = time.monotonic()
-        await asyncio.wait([async_function(), async_function()])
-        end = time.monotonic()
-        # It should take at least 2 seconds as there's only one worker.
-        assert end - start >= 2
-    finally:
-        loop.set_default_executor(old_executor)
+    monkeypatch.setattr(sync_to_async, "executor", ThreadPoolExecutor(max_workers=1))
+    async_function = sync_to_async(sync_function)
+    start = time.monotonic()
+    await asyncio.wait([async_function(), async_function()])
+    end = time.monotonic()
+    # It should take at least 2 seconds as there's only one worker.
+    assert end - start >= 2
+
+
+@pytest.mark.asyncio
+async def test_sync_to_async_ASGI_THREADS(monkeypatch):
+    def sync_function():
+        time.sleep(.5)
+        return 42
+
+    # Set workers to 1 via env, call it twice and make sure that works right
+    monkeypatch.setenv("ASGI_THREADS", 1)
+    monkeypatch.setattr(sync_to_async, "executor", None)
+
+    async_function = sync_to_async(sync_function)
+    assert async_function.executor._max_workers == 1
+    start = time.monotonic()
+    await asyncio.wait([async_function(), async_function()])
+    end = time.monotonic()
+    # It should take at least 1 second as there's only one worker.
+    assert end - start >= 1
+
+    # Uses existing executor instance on class.
+    monkeypatch.setenv("ASGI_THREADS", 99)
+    orig_executor = async_function.executor
+    async_function = sync_to_async(sync_function)
+    assert async_function.executor is orig_executor
```

Comment:
Not sure about this though: it might actually make sense to use a separate executor for every decorator?
This could also be achieved through a new kwarg.